### ISSUE TITLE
Inject taskScheduler bean it can't work

### ISSUE THREAD

**Issue by Fangfeikun** [lukas-krecan/ShedLock#93]
When i use follow code ,it can't work 
```
@Bean()
    public ThreadPoolTaskScheduler taskScheduler()
    {
        ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
        taskScheduler.setPoolSize(10);
        return taskScheduler;
    }

 @Bean
    public ScheduledLockConfiguration taskScheduler(@Qualifier("fetchResourceUsage") LockProvider lockProvider,TaskScheduler taskScheduler) {
        return ScheduledLockConfigurationBuilder
            .withLockProvider(lockProvider)
            .withTaskScheduler(taskScheduler)
            .withDefaultLockAtMostFor(Duration.ofMinutes(10))
            .build();
    }

but use this ,it works.
 @Bean
    public ScheduledLockConfiguration taskScheduler(@Qualifier("fetchResourceUsage") LockProvider lockProvider) {
        return ScheduledLockConfigurationBuilder
            .withLockProvider(lockProvider)
            .withPoolSize(10)
            .withDefaultLockAtMostFor(Duration.ofMinutes(10))
            .build();
    }
```

**Comment by Fangfeikun** [comment]
<dependency>
			<groupId>net.javacrumbs.shedlock</groupId>
			<artifactId>shedlock-spring</artifactId>
			<version>0.18.2</version>
		</dependency>

		<dependency>
			<groupId>net.javacrumbs.shedlock</groupId>
			<artifactId>shedlock-provider-jdbc</artifactId>
			<version>0.18.2</version>
		</dependency>

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. I am planning to release a version which simplifies the usage significantly so such issues will not happen

**Comment by Fangfeikun** [comment]
Thanks

**Comment by lukas-krecan** [comment]
Just released version 2.0.1 with simplified configuration
